tests/engine/test_arg_utils.py

```diff
@@ -115,9 +116,9 @@ class NestedConfig:
 class DummyConfig:
     regular_bool: bool = True
     """Regular bool with default True"""
-    optional_bool: bool | None = None
+    optional_bool: bool = Field(default=None)
     """Optional bool with default None"""
-    optional_literal: Literal["x", "y"] | None = None
+    optional_literal: Literal["x", "y"] = Field(default=None)
```

Comment:
I don't think we should change this (there might be other fields that DO allow None), but you can add a `default_bool` and `default_literal` with `Field(default=None)` and check proper semantics on those